linkerd/detect/src/lib.rs

```diff
@@ -43,6 +46,16 @@ pub struct DetectService<T, D, N> {
 
 const BUFFER_CAPACITY: usize = 1024;
 
+pub fn allow_timeout<P, T>((p, t): (Detected<P>, T)) -> (Option<P>, T) {
+    match p {
+        Ok(p) => (p, t),
+        Err(DetectTimeout(timeout)) => {
+            info!(?timeout, "Protocol detection timeout");
```

Comment:
very minor, just a potential nice-to-have: if the intention is eventually to use this for both HTTP and TLS detection, it could be useful to have this include _which_ protocol is being detected...maybe something like:
```suggestion
            info!(
                ?timeout,
                protocol = %std::any::type_name::<P>(),
                "Protocol detection timeout"
            );
```